# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1467835874 18000
#      Wed Jul 06 15:11:14 2016 -0500
# Node ID eb751495ba43e8a4180ecf1c3a8d6bbb9531073b
# Parent  ccf50f0299992c40caebb035135b4c23861110bd
Condition grid_idx assignment on existence of axis_idx (bug #48401).

* __gnuplot_draw_axes__.m: Move initialization of grid_idx to within the
  major/minor grid code and choose initial value based upon whether an axis
  was drawn.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1469,20 +1469,24 @@ function __gnuplot_draw_axes__ (h, plot_
             fputs (plot_stream, "unset border;\n");
           endif
         endif
         axis_idx = do_border_2d (axis_obj, plot_stream, axis_idx);
       endif
     endif
   endif
 
-  grid_idx = axis_idx;
   if (! have_major_grid && ! have_minor_grid)
     fputs (plot_stream, "unset grid;\n");
   else
+    if (exist ("axis_idx", "var"))
+      grid_idx = axis_idx;
+    else
+      grid_idx = data_idx;
+    endif
     if (have_major_grid)
       grid_idx += 1;
       grid_obj.linestyle = axis_obj.gridlinestyle;
       grid_obj.linewidth = axis_obj.linewidth;
       grid_obj.alpha = axis_obj.gridalpha;
       [style, sidx_major] = do_linestyle_command (grid_obj, axis_obj.gridcolor,
                                                   grid_idx, plot_stream);
     else
